python/lsst/analysis/tools/tasks/cpAnalysis.py

```diff
@@ -0,0 +1,361 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "CpAnalysisConfig",
+    "CpAnalysisCalibConfig",
+    "CpBiasAnalysisTask",
+    "CpDarkAnalysisTask",
+    "CpFlatAnalysisTask",
+    "CpPtcAnalysisTask",
+)
+
+import numpy as np
+from lsst.pipe.base import connectionTypes as cT
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class CpAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("instrument",),
+    defaultTemplates={"outputName": "cpTest1"},
+):
+    data = cT.Input(
+        doc="Input data from cp_verify.",
+        name="verifyCalibStats",
+        storageClass="StructuredDataDict",
+        dimensions=("instrument",),
+    )
+    rawData = cT.Input(
+        doc="Raw values that should be collated earlier.",
+        name="verifyCalibDetStats",
+        storageClass="StructuredDataDict",
+        dimensions=(
+            "instrument",
+            "exposure",
+            "detector",
+        ),
+        deferLoad=True,
+        multiple=True,
+    )
+
+    camera = cT.PrerequisiteInput(
+        doc="Input camera to use for focal plane geometry.",
+        name="camera",
+        storageClass="Camera",
+        dimensions=("instrument",),
+        isCalibration=True,
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+
+class CpAnalysisConfig(AnalysisBaseConfig, pipelineConnections=CpAnalysisConnections):
+    def setDefaults(self):
+        super().setDefaults()
+
+
+class CpAnalysisCalibConnections(
+    AnalysisBaseConnections,
```

Comment:
I think you should be able to make a `CpAnalysisBaseConnections` class with no `defaultTemplates` and have `CpAnalysisConnections`  and `CpAnalysisCalibConnections` inherit from that instead of `AnalysisBaseConnections` to avoid duplication, if they're going to share most of the connections and methods.